torch/_inductor/ir.py

```diff
@@ -1680,10 +1680,20 @@ def _normalize_size(x, new_size):
         old_size = x.get_size()
         old_size = [None] * (len(new_size) - len(old_size)) + list(old_size)
         assert len(new_size) == len(old_size)
+        is_unbacked = V.graph.sizevars.shape_env.is_unbacked_symint
         for i in range(len(new_size)):
             if new_size[i] == -1:
                 assert old_size[i] is not None
                 new_size[i] = old_size[i]
+            elif old_size[i] is None or old_size[i] == 1 or is_unbacked(old_size[i]):
+                pass
+            elif is_unbacked(new_size[i]):
+                # Assume equal, take backed symint
+                # TODO: This may be unsound
+                new_size[i] = old_size[i]
```

Comment:
You want to make a deferred runtime assert. You can trigger that with expect_true(new_size[i] == old_size[i]). Actually it's a bit strange `guard_equals` doesn't do this automatically